libdnf/dnf-context.cpp

```diff
@@ -847,6 +847,7 @@ HyGoal
 dnf_context_get_goal(DnfContext *context)
 {
     DnfContextPrivate *priv = GET_PRIVATE(context);
+    priv->goal->set_protect_running_kernel(libdnf::getGlobalMainConfig().protect_running_kernel().getValue());
```

Comment:
This is a bit weird, actually, to set the option on the goal in the "getter" (you'd be doing that multiple times if the getter gets called more than once and then you need to do this again in e.g. `dnf_context_run()`).

Can't you set this when the goal is created? That would seem to be the right place.

Reply:
Ok, thanks. I moved it to ``hy_goal_create`` and ``hy_goal_clone``.